airbyte-integrations/connectors/source-google-pagespeed-insights/source_google_pagespeed_insights/google_pagespeed_insights.yaml

```diff
@@ -0,0 +1,63 @@
+version: "0.1.0"
+definitions:
+  selector:
+    extractor:
+      field_pointer: []
+  requester:
+    url_base: "https://www.googleapis.com/pagespeedonline/v5/"
+    http_method: "GET"
+    request_options_provider:
+      request_parameters:
+        key: "{{ config['api_key'] }}"
+        url: "{{ config['url'] }}"
+        category: seo
+        cateogry: accessibility
+        cateogry: best-practices
+        cateogry: performance
+        cateogry: pwa
+        strategy: desktop
+        strategy: mobile
+  retriever:
+    record_selector:
+      $ref: "*ref(definitions.selector)"
+    paginator:
+      type: NoPagination
+    requester:
+      $ref: "*ref(definitions.requester)"
+  base_stream:
+    retriever:
+      $ref: "*ref(definitions.retriever)"
+  pagespeed_stream:
+    $ref: "*ref(definitions.base_stream)"
+    $options:
+      name: "pagespeed"
+      path: "/runPagespeed"
+  origin_pagespeed_stream:
```

Comment:
The "problem" I had when using one stream was, that the check - command takes so long. So, the "normal" request without "origin" - prefix takes between 20 and 40 seconds. But the integration tests time out after 30 seconds. I also thought, that it might be inconvenient for the user to wait for so long? The "origin" request however returns after some seconds.
Any suggestion on how to solve this? It would be awesome to use a different url (the one with "origin" just for the check, but without needing to create a separate stream. Is this possible somehow?

Reply:
This is odd! As far as I can see, both streams make the exact same request to the API I think, so they should take the same time?

Either way, if the 30 secondes itself is the issue, it can be raised!

You can update the `acceptance-test-config.yml` file (`connection` test, and add a bigger `timeout_seconds` property)
https://docs.airbyte.com/connector-development/testing-connectors/source-acceptance-tests-reference#test-connection

Reply:
Ok I'll increase the timeout than.

If you prefix the url with origin, only a very reduced (aggregates) report is generated and then returned. Therefore it takes way less time. See this faq here: https://developers.google.com/speed/docs/insights/faq?hl=en

I'll make this clearer in the docs

Reply:
Interesting! However the page (faq) you're mentioning seems deprecated, and the `origin:` prefix isn't mentioned in the new documentation, so I wouldn't rely on it!

Reply:
I've seen that as well. I tried it with postman however and with origin: you only get the main values (fcp, lcs, etc), whereas without origin we get the whole report.
But still I see that this is kinda confusing. I have a suggestion: as we remove the second stream anyhow, I could make the url list in a way that it not only allowa http(s)://xyz.com but also origin:http...
Then users can choose themselves. What do you think?

Reply:
Yes, this could work! 

I would put a huge warning in the documentation that this syntax may be deprecated at any time by google so that users are aware of the risks involved!